deps/chakrashim/src/v8trycatch.cc

```diff
@@ -78,23 +78,34 @@ void TryCatch::GetAndClearException() {
   }
 
   if (hasException) {
-    JsValueRef exceptionRef;
-    errorCode = JsGetAndClearException(&exceptionRef);
+    JsValueRef metadataRef;
+    errorCode = JsGetAndClearExceptionWithMetadata(&metadataRef);
     // We came here through JsHasException, so script shouldn't be in disabled
     // state.
     CHAKRA_ASSERT(errorCode != JsErrorInDisabledState);
     if (errorCode == JsNoError) {
-      error = exceptionRef;
+      metadata = metadataRef;
     }
   }
 }
 
 Handle<Value> TryCatch::ReThrow() {
-  if (error == JS_INVALID_REFERENCE) {
+  if (metadata == JS_INVALID_REFERENCE) {
     GetAndClearException();
   }
 
-  if (error == JS_INVALID_REFERENCE) {
+  if (metadata == JS_INVALID_REFERENCE) {
+    return Local<Value>();
+  }
```

Comment:
Admittedly I find this pattern to be unintuitive, seems like maybe the signature/name of the method could be updated to make it more clear?